bpf/bpf_sock.c

```diff
@@ -322,7 +322,12 @@ static __always_inline int __sock4_xlate_fwd(struct bpf_sock_addr *ctx,
 	if (!svc)
 		return -ENXIO;
 	if (svc->count == 0)
+#if defined(ENABLE_L7_LB)
+		if (!lb4_svc_is_l7loadbalancer(svc))
+			return -EHOSTUNREACH;
+#else
 		return -EHOSTUNREACH;
+#endif
```

Comment:
```suggestion
		if (!lb4_svc_is_l7loadbalancer(svc))
			return -EHOSTUNREACH;
```

It's ok to call `lb4_svc_is_l7loadbalancer()` when L7 LB is disabled. Same for IPv6.

Reply:
ah you are right, lb4_svc_is_l7loadbalancer already checks if ENABLE_L7_LB is enabled.